test/checkr/api_class_test.rb

```diff
@@ -48,6 +48,24 @@ class ApiClassTest < Test::Unit::TestCase
           :filter => 'test filter',
         })
       end
+
+      should 'allow passing an api_key as a param' do
+        response = test_response(test_mock_resource_list)
+        api_key = '123456'
+
+        @mock.expects(:get).with do |url, headers, params|
+          (url == "#{Checkr.api_base}#{MockResource.path}?page=1&filter=test%20filter" ||
+            url == "#{Checkr.api_base}#{MockResource.path}?filter=test%20filter&page=1") &&
+            params == nil &&
```

Comment:
@jperichon Here for example it checks that params are nil, so api_key was removed, and it also checks for the url to not include api_key as a query.